server/util.go

```diff
@@ -347,6 +348,21 @@ func WaitForQuitSignals() ErrorCode {
 	return ErrorCode{Code: int(sig.(syscall.Signal)) + 128}
 }
 
+// GetAppDBBackend gets the backend type to use for the application DBs.
+func GetAppDBBackend(opts types.AppOptions) dbm.BackendType {
+	rv := cast.ToString(opts.Get("app-db-backend"))
+	if len(rv) == 0 {
+		rv = sdk.DBBackend
+	}
+	if len(rv) == 0 {
+		rv = cast.ToString(opts.Get("db-backend"))
+	}
+	if len(rv) != 0 {
+		return dbm.BackendType(rv)
+	}
+	return dbm.GoLevelDBBackend
+}
+
```

Comment:
> what about introducing an sdk specific db config? its a bit more burden some to do it this way but right now im guessing most people are running rocks on the app and LevelDB on tendermint.

I just pushed up a change that added `app-db-backend` to the `app.toml` and config object. I created this `GetAppDBBackend` function to house the logic of deciding which db backend setting to use.

Order of precedence:
1. The new `app-db-backend` config entry in `app.toml`.
2. The deprecated `DBBackend` variable that is set during compilation.
3. Tendermint's `db-backend` config entry in `config.toml`.
4. A default of GoLevelDB.

This does a few nice things:
1. Provides a configuration file replacement for the compile-time `DBBackend` value.
2. Allows for a less troublesome upgrade by maintaining current db types. After the upgrade, node operators can update their `app.toml` config files at their leisure. By the upgrade after that, the compiled executable won't need to have `DBBackend` set anymore.
3. If only a single db backend type is desired, the Tendermint config's `db-backend` can apply to the Cosmos-SDK DBs too, providing a single config entry for all DBs.

I was tempted to make a magic `app-db-backend` value of `tendermint` to explicitly tell it to use the same value as `db-backend` (and to ignore `DBBackend`), but I hate making magic values, and it'd only really make sense to have it until `DBBackend` is fully removed.